language/vm/vm-runtime/src/code_cache/module_cache.rs

```diff
@@ -45,11 +61,27 @@ pub trait ModuleCache<'alloc> {
     /// * `Ok(FunctionRef)` if such function exists.
     /// * `Err(...)` for a verification issue in a resolved dependency, VM invariant violation, or
     ///   function not found.
-    fn resolve_function_ref(
+    pub fn resolve_function_ref(
         &self,
         caller_module: &LoadedModule,
         idx: FunctionHandleIndex,
-    ) -> VMResult<FunctionRef<'alloc>>;
+        data_view: &dyn InterpreterContext,
```

Comment:
I've been thinking of putting a `&dyn ModuleFetcher` and have `impl<T: InterpreterContext> ModuleFetcher for T`. But that doesn't seem to be how rust trait object system works. Thoughts?

Reply:
well I guess I made my comments above ;)
I am not clear we need a `ModuleFetcher` or an `InterpreterContext`.
Right now I put everything behind `InterpreterContext` because it was the simplest way to do this and  still the PR is massive.
I really think we should try to stay away from traits and lifetimes unless they are key to the code.